gc/policy/policy.go

```diff
@@ -0,0 +1,217 @@
+package policy
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/metadata"
+	"github.com/containerd/containerd/plugin"
+)
+
+type config struct {
+	PauseThreshold float64 `toml:"pause_threshold"`
+
+	ScheduleOnDelete  bool `toml:"on_delete"`
+	MutationThreshold int  `toml:"mutation_threshold"`
+
+	ScheduleDelayMs int `toml:"schedule_delay_ms"`
+	StartupDelayMs  int `toml:"startup_delay_ms"`
+}
+
+func init() {
+	plugin.Register(&plugin.Registration{
+		Type: plugin.GCPlugin,
```

Comment:
GCPlugin -> GCPolicyPlugin?

Reply:
This plugin happens to implement the policy, I think it is fine to refer to this as a GC plugin. I am not sure the type here has too much meaning since it is not exporting a type used by other systems. It is possible though that we will want this type to expose some sort of metrics and maybe a trigger mechanism. That interface I would imagine would be more about GC rather than the GC policy (stats about pause time, triggering, policy config).